generate_pr_summary.py

```diff
@@ -0,0 +1,68 @@
+import os
+import requests
+import json
+
+# Fetch environment variables
+PR_SUMMARY_TOKEN = os.getenv('PR_SUMMARY_TOKEN')
+OPENAI_API_KEY = os.getenv('OPENAI_API_KEY')
+
+# Define headers
+github_headers = {
+    'Authorization': f'token {PR_SUMMARY_TOKEN}',
+    'Accept': 'application/vnd.github.v3+json'
+}
+
+openai_headers = {
+    'Content-Type': 'application/json',
+    'Authorization': f'Bearer {OPENAI_API_KEY}'
+}
+
+# Function to fetch PR details
+def fetch_pr_details(pr_number, repo_owner, repo_name):
+    url = f'https://api.github.com/repos/{repo_owner}/{repo_name}/pulls/{pr_number}'
+    response = requests.get(url, headers=github_headers)
+    return response.json()
+
+# Function to generate summary using OpenAI
+def generate_summary(pr_details):
+    prompt = f"Summarize the following PR details for documentation: {pr_details}"
+    data = {
+        "model": "text-davinci-003",
+        "prompt": prompt,
+        "max_tokens": 150
+    }
+    response = requests.post("https://api.openai.com/v1/completions", headers=openai_headers, data=json.dumps(data))
+    return response.json()['choices'][0]['text']
+
+# Function to post summary as a comment on the issue
+def post_summary_to_issue(issue_number, summary, repo_owner, repo_name):
```

Comment:
Test
```suggestion
This is the suggestion you want to make. It may be changing a word or two, correcting grammar, or restructuring the text completely.
```